### PR TITLE
[IMP] product: archive should unpublish product

### PR DESCRIPTION
**Pull request by kea14**
On archive of product template, it also unpublishes product template.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_sale/models/product.py [hunk 1]
```diff
@@ -392,6 +392,12 @@ def _get_images(self):
         self.ensure_one()
         return [self] + list(self.product_template_image_ids)
 
+    def action_archive(self):
+        values = {}
+        if 'website_published' in self.env['product.template']._fields:
+            values['website_published'] = not self.website_published
+        self.write(values)
+        return super(ProductTemplate, self).action_archive()
```
Comment (qsm-odoo) [inline addons/website_sale/models/product.py]: Your code says:
"when archiving, check that products templates have a website_published field (is not this always true??), and then change its value"... come on... don't tell me you tested this one...

And even so:
- `self` is a multi-records recorset so you cannot read fields out of it like that, try archiving two products -> crash.
- `self.env['product.template']._fields` == `self._fields`
- ...

Comment (kea14) [reply addons/website_sale/models/product.py]: Tbh I had a look at the code of `/website/publish` route. I'll check this out, thx :smiley:

Comment (kea14) [reply addons/website_sale/models/product.py]: I modified this code, it seems to be better and work when archiving 2+ products

Comment (qsm-odoo) [reply addons/website_sale/models/product.py]:
```python
def action_archive(self):
    self.write({'website_published': False)
    return super(ProductTemplate, self).action_archive()
```

**Pull request closed without merging**